manifests/init.pp

```diff
@@ -30,32 +40,53 @@
 #   *If* the $servers array above starts with '127.0.0.1' or '::1', then
 #   the system will set itself up as a caching nameserver unless this is set
 #   to false.
-#
+# @param sortlist
+#   Optional Array of address/netmask pairs that allow addresses returned by
+#   gethostbyname to be sorted.
+# @param extra_options
+#   Optional Array to put any options that may not be covered by the variables
+#   below. These will be appended to the options string.
 class resolv (
   Simplib::Netlist           $servers        = simplib::lookup('simp_options::dns::servers', { 'default_value' => ['127.0.0.1'] }),
```

Comment:
The type for $servers is erroneously more general than the EPP allows for its corresponding parameter, $nameservers.  Shouldn't the two match? If someone attempts to put in a hostname, won't the EPP barf, anyways?  If so, backward compatibility is not achieved, but shouldn't be expected.

Reply:
I had considered this (and agree).  The type was initially left as-is to avoid the assumption of a breaking API change―but as you've noticed, the EPP template will not permit invalid data.